src/widgets/Updater.cpp

```diff
@@ -661,6 +661,9 @@ bool Updater::TickCustom(wxDC& dc, int labelIdx, wxFont font,
 }
 
 // Theoretically this helps
+
+Updater::~Updater() {}
+
```

Comment:
The third commit corrects the omission of the virtual destructor.  Good:  but, logically this should be squashed with the same commit that introduces 

```
std::unique_ptr<Updater> mpUpdater;
```

in Ruler.h, so the known mistake of missing default constructors is never even introduced into the commit history.

Besides that, I recommend:
- Define all of the destructors in this commit as non-inline, not only the one for Updater
- You can use the C++11 alternative syntax: `Updater::~Updater() = default`
- Also, now that I think of this too:  when you introduce the subclasses of `Updater`, make them `final` if possible, e.g. `struct LinearUpdater final : public Updater { ...` this means: you do not intend further subclassing of those classes.

What are my reasons for these practices?
- A linker has to put the virtual function table for each class in some unique place, and often implemented to follow this rule: the .cpp file definining the first non-inline virtual function generates the object file in which the table will live.  But, if there is non non-inline virtual function, then the compiler has fall back on generating multiple copies of the virtual function table that the linker just eliminates later.  Moral:  be nice to your linker, and avoid in-line virtuals.
- in-line optimization is inherently incompatible with virtual function dispatch anyway, so what do it ? ...
- except: when a class is `final`, and the declared object is of the leaf-class type, then the compiler can know that it really can to do the inline optimization, and may be smart enough to in-line default destructors.